pkg/kubelet/images/image_gc_manager.go

```diff
@@ -113,22 +113,29 @@ type imageCache struct {
 	images []container.Image
 }
 
-// set updates image cache.
+// set sorts the input list and updates image cache.
+// 'i' takes ownership of the list, you should not reference the list again
+// after calling this function.
 func (i *imageCache) set(images []container.Image) {
 	i.Lock()
 	defer i.Unlock()
+	// Make a copy of the input list. This doesn't completely avoid
+	// potential data race. The caller should not reference the list again
+	// after calling this function.
```

Comment:
I think this comment is unnecessarily alarming... the set() method is private, and the only place we call it does not hold onto the list and change it further. The godoc on the method clarifying ownership is probably sufficient.

Reply:
thanks. Removed the comment